pandas/core/internals/blocks.py

```diff
@@ -632,7 +633,10 @@ def _astype(self, dtype, copy=False, errors='raise', values=None,
             return self
 
         if klass is None:
-            if dtype == np.object_:
+            if is_sparse(self.values):
+                # Series[Sparse].astype(object) is sparse.
+                klass = ExtensionBlock
+            elif is_object_dtype(dtype):
                 klass = ObjectBlock
             elif is_extension_array_dtype(dtype):
```

Comment:
so maybe should just move the ``is_extension_array_dtype`` up to first, and add a ``is_extension_dtype(self.values)`` test as well (should encompas your is_sparse check) and is more general

Reply:
I'll make that change and run the test suite.

I was kinda worried about "false positives" here, but I suppose it's exactly what we want if an extension array claims it's object dtype.

Reply:
As posted in the unstack PR, we need to special case Space here, since it's the only (internal) extension type that has special `.astype(object)` behavior.